xdem/coreg.py

```diff
@@ -596,6 +593,90 @@ def to_matrix(self) -> np.ndarray:
+
+        :param reference_dem: 2D array of elevation values acting reference. 
+        :param dem_to_be_aligned: 2D array of elevation values to be aligned.
+        :param inlier_mask: Optional. 2D boolean array of areas to include in the analysis (inliers=True).
+        :param transform: Optional. Transform of the reference_dem. Mandatory in some cases.
+
+        :returns: A 1D array of finite residuals.
+        """
+        # Use the transform to correct the DEM to be aligned.
+        aligned_dem = self.apply(dem_to_be_aligned, transform=transform)
+
+        # Format the reference DEM
+        ref_arr, ref_mask = xdem.spatial_tools.get_array_and_mask(reference_dem)
+
+        if inlier_mask is None:
+            inlier_mask = np.ones(ref_arr.shape, dtype=bool)
+
+        # Create the full inlier mask (manual inliers plus non-nans)
+        full_mask = (~ref_mask) & np.isfinite(aligned_dem) & inlier_mask
+
+        # Calculate the DEM difference
+        diff = ref_arr - aligned_dem
+
+        # Return the difference values within the full inlier mask
+        return diff[full_mask]
+
+    def error(self, reference_dem: Union[np.ndarray, np.ma.masked_array],
+              dem_to_be_aligned: Union[np.ndarray, np.ma.masked_array],
+              error_type: str = "nmad",
+              inlier_mask: Optional[np.ndarray] = None,
+              transform: Optional[rio.transform.Affine] = None) -> float:
+        """
+        Calculate the error of a coregistration approach.
+
+        Choices:
+            - "nmad": Default. The Normalized Median Absolute Deviation of the residuals.
+            - "median": The median of the residuals.
+            - "mean": The mean/average of the residuals
+            - "std": The standard deviation of the residuals.
+            - "rms": The root mean square of the residuals.
+
+        :param reference_dem: 2D array of elevation values acting reference. 
+        :param dem_to_be_aligned: 2D array of elevation values to be aligned.
+        :param error_type: The type of error meaure to calculate.
+        :param inlier_mask: Optional. 2D boolean array of areas to include in the analysis (inliers=True).
+        :param transform: Optional. Transform of the reference_dem. Mandatory in some cases.
+
+        :returns: The error measure of choice for the residuals.
+        """
+
+        residuals = self.residuals(reference_dem=reference_dem, dem_to_be_aligned=dem_to_be_aligned,
+                                   inlier_mask=inlier_mask, transform=transform)
+
+        if error_type == "nmad":
+            error = xdem.spatial_tools.nmad(residuals)
+        elif error_type == "median":
+            error = np.median(residuals)
+        elif error_type == "mean":
+            error = np.mean(residuals)
+        elif error_type == "std":
+            error = np.std(residuals)
+        elif error_type == "rms":
```

Comment:
we could add the Mean Absolute Error, less sensitive to outliers (https://medium.com/human-in-a-machine-world/mae-and-rmse-which-metric-is-better-e60ac3bde13d)

Reply:
Added in 43b1d3e